frontend/components/AuthGate/AuthGate.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ReactElement } from 'react'
+import { useRouter } from 'next/router'
+import LoadingWrapper from '../LoadingWrapper'
+import { useCurrentUserQuery } from '../../generated/graphql'
+
+/**
+ * Checks that the user is logged in.
+ * If not, redirects them to the login page.
+ */
+
+type AuthGateProps = {
+  children: ReactElement
```

Comment:
nit: it's usually best to default to the `ReactNode` type since it encompasses more possible values that we'd want to use potentially:

```js
type ReactChild = ReactElement | ReactText;
type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```

Reply:
VSC yelled when we tried that, but maybe because LoadingWrapper also expects ReactElement, if we can satisfy TSC with ReactNode then that's good with me

Reply:
Yeah should be as simple as just updating the type definition in `LoadingWrapper`. We had it as `ReactNode` first but this was just an effort to avoid updating too much stuff, I can go and update that and it should work

Reply:
Actually, changing it to `ReactNode` in `LoadingWrapper` does indeed seem to break things and cause VS Code to shout. We can look into that more later if we want but I would advocate for not holding this PR up over that now.